src/benchmarks/micro/coreclr/BilinearInterpol/BilinearInterpol.cs

```diff
@@ -0,0 +1,315 @@
+                Vector128<int> a = Avx.ConvertToVector128Int32(aDouble);
+                a = Sse41.Min(Sse41.Max(a, Vector128<int>.Zero), ALengthMinusOne);
+                Vector128<int> aPlusOne = Sse41.Min(Sse2.Add(a, One), ALengthMinusOne);
+
+                // Now, get the reference input, xA, for our index a.
+                // This involves casting from  int to double.
+                Vector256<double> xA = Avx.Add(Avx.Multiply(Avx.ConvertToVector256Double(a), vDeltaA), vMinXA);
+
+                // Now, compute the lambda for our A reference point.
+                Vector256<double> currentXNormA = Avx.Max(vMinXA, Avx.Min(currentX, vMaxXA));
+                Vector256<double> lambdaA = Avx.Multiply(Avx.Subtract(currentXNormA, xA), vInvDeltaA);
+
+                // Now, we need to load up our reference points using Vector Gather operations.
+                Vector256<double> AVector = Avx2.GatherVector256(pA, a, 8);
+                Vector256<double> AVectorPlusOne = Avx2.GatherVector256(pA, aPlusOne, 8);
+
+                // Now, do the all of the above for our B reference point.
+                Vector256<double> bDouble = Avx.Multiply(Avx.Subtract(currentX, vMinXB), vInvDeltaB);
+                Vector128<int> b = Avx.ConvertToVector128Int32(bDouble);
+                b = Sse41.Min(Sse41.Max(b, Vector128<int>.Zero), BLengthMinusOne);
+                Vector128<int> bPlusOne = Sse41.Min(Sse2.Add(b, One), BLengthMinusOne);
+
+                Vector256<double> xB = Avx.Add(Avx.Multiply(Avx.ConvertToVector256Double(b), vDeltaB), vMinXB);
+                Vector256<double> currentXNormB = Avx.Max(vMinXB, Avx.Min(currentX, vMaxXB));
+                Vector256<double> lambdaB = Avx.Multiply(Avx.Subtract(currentXNormB, xB), vInvDeltaB);
+
+                Vector256<double> BVector = Avx2.GatherVector256(pB, b, 8);
+                Vector256<double> BVectorPlusOne = Avx2.GatherVector256(pB, bPlusOne, 8);
+
+                Vector256<double> newZ = Avx.Add(Avx.Multiply(vWeightA, Avx.Add(AVector, Avx.Multiply(lambdaA, Avx.Subtract(AVectorPlusOne, AVector)))),
+                                             Avx.Multiply(vWeightB, Avx.Add(BVector, Avx.Multiply(lambdaB, Avx.Subtract(BVectorPlusOne, BVector)))));
+                Avx.Store(pZ, newZ);
+            }
+        }
+        return z;
+    }
+#endif // !NETCOREAPP2_1 && !NETCOREAPP2_2 && !NETFRAMEWORK
+
+    public static bool CheckResult(double[] output, double[] vectorOutput)
+    {
+        double eps = 1e-16;
+        for (int i = 0; i < output.Length; i++)
+        {
+            if (Math.Abs(output[i] - vectorOutput[i]) > eps)
+            {
+                Console.WriteLine("Failed at " + i + ": output is " + output[i] + ", vectorOutput is " + vectorOutput[i]);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    [Benchmark(Description = "BilinearInterpol_Scalar")]
```

Comment:
we use `Description` to set the display name for benchmarks which can not be renamed (method name is part of the [benchmark id](https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#benchmarks-are-immutable))

in this case we could just name the  benchmark `Interpol_Vector` (keyword `Bilinear` is already used in the type name which is also part of benchmark id)

```suggestion
    [Benchmark]
```